### DISCUSSION TITLE
Generating subkeys for regtest falls back to bitcoin mainnet format

### DISCUSSION THREAD

**Discussion by STCB** [1200wd/bitcoinlib#404] [Q&A] [answered]
Hi there !

I'm trying to generate addresses in regtest, using `HDKey.subkey_for_path()` function, with a custom path and 'regtest' as network, but this silently generates me addresses with a mainnet format.
I tested the function with 'testnet' rather than 'regtest', and the testnet format is good.

I've seen in keys.py/check_network_and_key that the network fallsback to default if the one provided is not handled.

Alright, from there I've been experimenting 20 minutes and found out that HDKey.from_seed(), which I use to generate my "master key", is also in a mainnet format.
So I tried using the CLI, to see if it was my usage or the lib that did not behave as I expected, and it appears that creating a new segwit address in regtest from the CLI does also get me a mainnet formatted address.

I'm really no expert in regtest and not much experimented yet, but when trying to generate coins from bcoin-cli running regtest with
`bcoin-cli rpc generatetoaddress 1 <bitcoinlib mainnet format generated address>` it tells me "Invalid address", which I understand as "this address is mainnet and not regtest".

**Comment by mccwdev** [comment, marked as answer]
I can reproduce this problem, can you show which code leads to this error.

Like this it works for me:

``` python
>>> phrase = 'table medal solution fit 
>>> k = HDKey.from_seed(phrase, network
>>> k.subkey_for_path('m/0/0/0/0')     
>>> print(k.network, k.address())      
<Network: regtest> bcrt1qp8krwvz78eumjnhkqk22pjyt4nh34nm65fuwdt
```

**Reply by STCB** [reply]
```
    data.master_key = HDKey.from_seed(data.conf.shared_seed, network=network, witness_type="segwit")
    print("Master key generated successfully.\n\tAddress:", data.master_key.address(), "\n\tNetwork:", network)
```

This code leads to this output:

```
Master key generated successfully.
        Address: bc1qx7fj8ym4wmuxmgrlxmg5jjn6nmke5k67xwvjvv 
        Network: regtest
```

And then, this code
```
    print("Deriving wallet at path", path, " and network", network)
    wallet = data.master_key.subkey_for_path(path, network=network)
    print("Derivated address:", wallet.address())
```

leads to this output:
```
Deriving wallet at path m/44'/0'/0'/0/0  and network regtest
Derivated address: bc1qecm2r6u2twlvj3tg0m7eqet0z4ndhpkvt03rg9
```

**Comment by mccwdev** [comment]
Not sure what is happening in the 'data' class.
But if I run the code like this it works:

```
shared_seed = 'table medal solution fit rice whale wrestle room network zero visit field'
network = 'regtest'
master_key = HDKey.from_seed(shared_seed, network=network, witness_type="segwit")
print("Master key generated successfully.\n\tAddress:", master_key.address(), "\n\tNetwork:", network)
path = "m/44'/0'/0'/0/0"
print("Deriving wallet at path", path, " and network", network)
wallet = master_key.subkey_for_path(path, network=network)
print("Derivated address:", wallet.address())

# Master key generated successfully.
# 	Address: bcrt1qp8krwvz78eumjnhkqk22pjyt4nh34nm65fuwdt 
# 	Network: regtest
# Deriving wallet at path m/44'/0'/0'/0/0  and network regtest
# Derivated address: bcrt1qv6ne5c6hnhdjf50yq2f4wpr00ugzp0st4upj4v
# Execution took 0.009 seconds
```

**Reply by STCB** [reply]
Alright, then it must be a config issue or something related to my env (windows 11):

```
python .\temp.py
Master key generated successfully.
        Address: bc1qp8krwvz78eumjnhkqk22pjyt4nh34nm6ux7sp3
        Network: regtest
Deriving wallet at path m/44'/0'/0'/0/0  and network regtest
Derivated address: bc1qv6ne5c6hnhdjf50yq2f4wpr00ugzp0stanrvek

```

That's what I got with your exact code !

In bitcoinlib logs, it tells me that I don't have fastecdsa installed and using a slower one, idk if this could be related...
Maybe I should try reinstalling bitcoinlib ?

**Reply by STCB** [reply]
So I did reinstall bitcoinlib, but it first made no changes, I had to delete the ~/.bitcoinlib and reinstall it to work it fine.
Great !
Even if I do not remember changing anything in there except for the providers.json to add Bcoin...
Thanks for your help !
